lib/promoted/ruby/client/options.rb

```diff
@@ -166,25 +168,25 @@ def request_params include_insertion: true
         end
 
         def compact_insertions
-          @compact_insertions = []
-          insertions_to_compact = full_insertion
-          paging = request[:paging] || {}
-          size = paging[:size]
-          unless size.nil? || size == 0
-            insertions_to_compact = insertions_to_compact[0..size-1]
-          end
-          from = paging[:from].to_i
+          @insertion            = [] # insertion should be set according to the compact insertion
+          paging                = request[:paging] || {}
+          size                  = paging[:size] ? paging[:size].to_i : 0
+          from                  = paging[:from].to_i
```

Comment:
I'm renaming 'from' in a PR to 'offset'.  We found out that having a proto field that matches a keyword in python causes issues.